homeassistant/components/august/__init__.py

```diff
@@ -352,6 +359,12 @@ def update_lock_status(self, lock_id, lock_status, update_start_time_utc):
         self._lock_last_status_update_time_utc_by_id[lock_id] = update_start_time_utc
         return True
 
+    def lock_has_doorsense(self, lock_id):
+        """Determine if a lock has doorsense installed and can tell when the door is open or closed."""
+        # We do not update here since this is not expected
+        # to change until restart
+        return self._lock_detail_by_id.get(lock_id).doorsense
```

Comment:
`.get()` can return `None`. If you know that it always exists, use `[]` which will raise if the key is not found.

```suggestion
        return self._lock_detail_by_id[lock_id].doorsense
```

Reply:
Based on `_update_locks_detail`, the value is set to `None` if a request error occurs. That would cause an attribute error. You should check if the lock detail is not None.